src/Tasks/Microsoft.NET.Build.Tasks/CompileReadyToRunImages.cs

```diff
@@ -0,0 +1,437 @@
+            string supportedRuntimeIdentifiers = frameworkRef == null ? null : frameworkRef.GetMetadata("RuntimePackRuntimeIdentifiers");
+
+            // Get information on the runtime package used for the current target
+            ITaskItem frameworkPack = RuntimePacks.Where(pack => pack.ItemSpec.EndsWith(".Microsoft.NETCore.App", StringComparison.InvariantCultureIgnoreCase)).SingleOrDefault();
+            _runtimeIdentifier = frameworkPack == null ? null : frameworkPack.GetMetadata(MetadataKeys.RuntimeIdentifier);
+            _packagePath = frameworkPack == null ? null : frameworkPack.GetMetadata(MetadataKeys.PackageDirectory);
+
+            var runtimeGraph = new RuntimeGraphCache(BuildEngine4, Log).GetRuntimeGraph(RuntimeGraphPath);
+            var supportedRIDsList = supportedRuntimeIdentifiers == null ? Array.Empty<string>() : supportedRuntimeIdentifiers.Split(';');
+
+            // Get the best RID for the host machine, which will be used to validate that we can run crossgen for the target platform and architecture
+            string hostRuntimeIdentifier = NuGetUtils.GetBestMatchingRid(
+                runtimeGraph, 
+                DotNet.PlatformAbstractions.RuntimeEnvironment.GetRuntimeIdentifier(), 
+                supportedRIDsList, 
+                out bool wasInGraph);
+
+            if (hostRuntimeIdentifier == null || _runtimeIdentifier == null || _packagePath == null)
+            {
+                Log.LogError(Strings.ReadyToRunNoValidRuntimePackageError);
+                return false;
+            }
+
+            if (!ExtractTargetPlatformAndArchitecture(_runtimeIdentifier, out string targetPlatform, out _targetArchitecture) ||
+                !ExtractTargetPlatformAndArchitecture(hostRuntimeIdentifier, out string hostPlatform, out Architecture hostArchitecture) ||
+                targetPlatform != hostPlatform)
+            {
+                Log.LogError(Strings.ReadyToRunTargetNotSuppotedError);
+                return false;
+            }
+
+            if (!GetCrossgenComponentsPaths() || !File.Exists(_crossgenPath) || !File.Exists(_clrjitPath))
+            {
+                Log.LogError(Strings.ReadyToRunTargetNotSuppotedError);
+                return false;
+            }
+
+            LogStandardErrorAsError = true;
+
+            // Run Crossgen on input assemblies
+            ProcessInputFileList(FilesToPublishPreserveNewest, _r2rPublishPreserveNewest);
+            ProcessInputFileList(FilesToPublishAlways, _r2rPublishAlways);
+
+            return !_crossgenFailures;
+        }
+
+        void ProcessInputFileList(ITaskItem[] inputFiles, List<ITaskItem> outputFiles)
+        {
+            if (inputFiles == null)
+            {
+                return;
+            }
+
+            // There seems to be duplicate entries of the same files in the input lists. Unify them
```

Comment:
I think this was a bug that was fixed. Is this still needed?

Reply:
Ah, let me check

Reply:
All good. I removed the workaround.